openwisp_monitoring/device/tests/__init__.py

```diff
@@ -51,23 +51,19 @@ def create_test_adata(self, no_resources=False):
         dd = DeviceData(pk=d.pk)
         self.assertDictEqual(dd.data, data)
         if no_resources:
-            metric_count, chart_count = 6, 4
+            metric_count, chart_count = 4, 4
         else:
-            metric_count, chart_count = 9, 7
+            metric_count, chart_count = 7, 7
         self.assertEqual(Metric.objects.count(), metric_count)
         self.assertEqual(Chart.objects.count(), chart_count)
         if_dict = {'wlan0': data['interfaces'][0], 'wlan1': data['interfaces'][1]}
         for ifname in ['wlan0', 'wlan1']:
             iface = if_dict[ifname]
-            for field_name in ['rx_bytes', 'tx_bytes']:
-                m = Metric.objects.get(
-                    key=ifname, field_name=field_name, object_id=d.pk
-                )
-                points = m.read(limit=10, order='time DESC')
-                self.assertEqual(len(points), 1)
-                self.assertEqual(
-                    points[0][m.field_name], iface['statistics'][field_name]
-                )
+            m = Metric.objects.get(key=ifname, field_name='rx_bytes', object_id=d.pk)
+            points = m.read(limit=10, order='time DESC', extra_fields=['tx_bytes'])
```

Comment:
Yes, I think it's InfluxDB specific but we might need something similar for the `read` method.
I mean we would definitely need to read metric data using params similar to `limit` and `order` for the other timeseries db to create required query.


* I think we can rename these params.
* have a list of keywords like [time, DESC, ASC, value, etc.] for order, and while working on #66, create queries for the other db based upon it.

Both are best done when working on #66, I think.

Reply:
We could do it kind like django does it: `order='time'` would translate in influxdb to `time ASC` while `order='-time'` would translate to `time DESC`. `limit` can stay. Regarding `extra_fields`, would that be addressed by the other PR which introduces metric configuration?

Rather than working on it in #66, I would start opening a new PR for this change, which being smaller it will be faster for me and @PabloCastellano to review and get merged.